### PR TITLE
test: Add attestation marshal tests

### PR DESCRIPTION
**Pull request by KirtiRamchandani**
Problem
Artifact attestation payload types are shared by organization, repository, and user attestation APIs, but the shared resource structs did not have dedicated JSON marshal coverage tracked by #55.

Root cause
Existing service tests exercise API decoding paths, while the resource-level marshal round-trip tests skipped Attestation and AttestationsResponse.

Solution
Add marshal tests for empty and populated Attestation and AttestationsResponse values. The populated cases use the existing json.RawMessage comparator so Sigstore bundle JSON is compared semantically.

Tests run
- gofmt -w github/attestations_test.go
- go test ./github -run 'TestAttestation(sResponse)?_Marshal'
- go test ./github
- go test ./...
- git diff --cached --check

Linked issue
Part of #55

### CODE REVIEW

File: github/attestations_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 The go-github AUTHORS. All rights reserved.
```
Comment (gmlewis) [inline github/attestations_test.go]:
```suggestion
// Copyright 2026 The go-github AUTHORS. All rights reserved.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Comment by KirtiRamchandani** [comment]
Thanks, updated the new test file copyright year to 2026 and reran the focused marshal test plus diff check.

File: github/attestations_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+package github
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestAttestation_Marshal(t *testing.T) {
+	testJSONMarshalOnly(t, &Attestation{}, `{"bundle": null, "repository_id": 0}`)
```
Comment (Not-Dhananjay-Mishra) [inline github/attestations_test.go:14]:
```suggestion
func TestAttestation_Marshal(t *testing.T) {
	t.Parallel()
	testJSONMarshalOnly(t, &Attestation{}, `{"bundle": null, "repository_id": 0}`)
```

File: github/attestations_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+			"mediaType": "application/vnd.dev.sigstore.bundle.v0.3+json"
+		},
+		"repository_id": 1
+	}`
+
+	testJSONMarshal(t, u, want, cmpJSONRawMessageComparator())
+}
+
+func TestAttestationsResponse_Marshal(t *testing.T) {
+	testJSONMarshal(t, &AttestationsResponse{}, `{"attestations": null}`)
```
Comment (Not-Dhananjay-Mishra) [inline github/attestations_test.go:32]:
```suggestion
func TestAttestationsResponse_Marshal(t *testing.T) {
	t.Parallel()
	testJSONMarshal(t, &AttestationsResponse{}, `{"attestations": null}`)
```

**Review by alexandear — COMMENTED** [review]
No need for `Attestation` and `AttestationsResponse` JSON marshal tests. We should only add tests for types with custom `MarshalJSON` or `UnmarshalJSON` implementations. For example:

- https://github.com/google/go-github/blob/d40cb978086391bed8115ca709affc454b9a5bee/github/repos_environments.go#L156
- https://github.com/google/go-github/blob/d40cb978086391bed8115ca709affc454b9a5bee/github/projects.go#L205

@gmlewis what do you think? These tests, generated by an LLM, don't actually validate anything meaningful and only add maintenance overhead.

**Comment by gmlewis** [comment]
> No need for `Attestation` and `AttestationsResponse` JSON marshal tests. We should only add tests for types with custom `MarshalJSON` or `UnmarshalJSON` implementations. For example:
> 
> * https://github.com/google/go-github/blob/d40cb978086391bed8115ca709affc454b9a5bee/github/repos_environments.go#L156
> * https://github.com/google/go-github/blob/d40cb978086391bed8115ca709affc454b9a5bee/github/projects.go#L205
> 
> @gmlewis what do you think? These tests, generated by an LLM, don't actually validate anything meaningful and only add maintenance overhead.

@alexandear - that's a good point. I'm wondering if #55 should simply be closed now and we should stop "advertising" this open-ended invitation for PRs.  This one issue was the sole target for previous Hacktoberfest contributions, and maybe it is no longer really needed.  Thoughts?

**Comment by alexandear** [comment]
> This one issue was the sole target for previous Hacktoberfest contributions, and maybe it is no longer really needed. Thoughts?

I vote for closing the issue.

Also, we can simply write a tool to generate marshal/unmarshal tests. But I'm not sure if we need.

**Comment by gmlewis** [comment]
> > This one issue was the sole target for previous Hacktoberfest contributions, and maybe it is no longer really needed. Thoughts?
> 
> I vote for closing the issue.
> 
> Also, we can simply write a tool to generate marshal/unmarshal tests. But I'm not sure if we need.

Agreed.  Thank you, @alexandear, and thank you, @KirtiRamchandani for your understanding. Closing.
